oracle/OracleBillDownload.py

```diff
@@ -89,15 +90,14 @@
 for key in my_dict.keys():
   destFilename = os.path.join(key, "concatenated.csv.gz")
   concatenatedFiles.append(destFilename)
-  bufferSize = 8  # Adjust this according to how "memory efficient" you need the program to be.
 
-  with open(destFilename, 'wb') as destFile:
+  with gzip.open(destFilename, 'wb') as destFile:
+    counter = 0
     for fileName in my_dict[key]:
-      with open(fileName, 'rb') as sourceFile:
-        chunk = True
-        while chunk:
-            chunk = sourceFile.read(bufferSize)
-            destFile.write(chunk)
+      with gzip.open(fileName, 'rb') as sourceFile:
+        for chunk in sourceFile.readlines()[counter:]:
```

Comment:
`readlines()` reads all the lines in memory; that's probably OK if the files are relatively small (up to a 100 MBs, say).

Reply:
I also worry about it keeping that whole gzip in memory, but I am hoping to scale the function app to 1G, etc.